crypto/evp/digest.c

```diff
@@ -51,7 +36,6 @@ void evp_md_ctx_clear_digest(EVP_MD_CTX *ctx, int force, int keep_fetched)
      * Don't assume ctx->md_data was cleaned in EVP_Digest_Final, because
      * sometimes only copies of the context are ever finalised.
      */
-    cleanup_old_md_data(ctx, force);
```

Comment:
I doubt about this change. See https://github.com/openssl/openssl/issues/29396

Reply:
`md_data` is a purely legacy thing. Actually what is missing in this PR is the removal of `md_data` member from the `evp_md_ctx_st`. @mattcaswell